### PR TITLE
Redstone related mappings and refactors

### PR DESCRIPTION
**Pull request by SpaceWalkerRS**
In terms of new mappings:
- `RedstonePowerView ` is an interface that extends `BlockView` and contains only methods for querying emitted and received redstone power. I went with the Yarn convention for names of these methods, even though they're a bit bulky, they're more consistent with all the redstone related names used elsewhere.

In terms of refactors:
- `AbstractRedstoneGateBlock` has been renamed to `RedstoneDiodeBlock` for a more accurate and concise name. Several methods in this class, as well as its subclasses `RepeaterBlock` and `ComparatorBlock`, have been renamed to better reflect whether they deal with input or output.
- `AbstractBlock#emitsRedstonePower` has been renamed to `isRedstonePowerSource` to make it less confusing.

### CODE REVIEW

**Review by OroArmor — APPROVED** [review]

**Review by eristhea — APPROVED** [review]
all the javadoc is lovely!

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 1]
```diff
@@ -15,6 +15,11 @@ CLASS net/minecraft/unmapped/C_triydqro net/minecraft/block/AbstractBlock
 	METHOD <init> (Lnet/minecraft/unmapped/C_triydqro$C_xnkxsdfy;)V
 		ARG 1 settings
 	METHOD m_bhohbmwk getWeakRedstonePower (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)I
+		COMMENT Returns the weak redstone power this block emits in the given direction.
+		COMMENT
```
Comment (EnnuiL) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 2]
```diff
@@ -202,6 +207,11 @@ CLASS net/minecraft/unmapped/C_triydqro net/minecraft/block/AbstractBlock
 		ARG 4 projectile
 	METHOD m_xfdklwdd getVerticalModelOffsetMultiplier ()F
 	METHOD m_ydisyayk getStrongRedstonePower (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)I
+		COMMENT Returns the strong redstone power this block emits in the given direction.
+		COMMENT
```
Comment (EnnuiL) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 3]
```diff
@@ -0,0 +1,48 @@
+CLASS net/minecraft/unmapped/C_gpzjbzcw net/minecraft/world/RedstonePowerView
+	FIELD f_caqhuckp DIRECTIONS [Lnet/minecraft/unmapped/C_xpuuihxf;
+	METHOD m_dnkvwchb getEmittedRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)I
+		COMMENT Returns the redstone power emitted from the given position in the given direction.
+		COMMENT This is the highest value between power emitted by the block itself, and strong redstone power
+		COMMENT received from neighboring blocks if the block is a redstone conductor.
+		COMMENT
```
Comment (EnnuiL) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		COMMENT <p>
+		COMMENT NOTE: directions in redstone power related methods are backwards, so this method
+		COMMENT checks for power in the <i>opposite</i> direction of the one given.
+		ARG 1 pos
+		ARG 2 dir
+	METHOD m_dsckesgu isEmittingRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)Z
+		COMMENT Returns whether redstone power is emitted from the given position in the given direction.
+		COMMENT
```
Comment (EnnuiL) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+		ARG 1 pos
+	METHOD m_hizjdvsd getReceivedRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;)I
+		COMMENT Returns the redstone power the given position receives from neighboring blocks.
+		ARG 1 pos
+	METHOD m_rgxdlzfx getReceivedStrongRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;)I
+		COMMENT Returns the strong redstone power the given position receives from neighboring blocks.
+		ARG 1 pos
+	METHOD m_uegtusaf getEmittedStrongRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)I
+		COMMENT Returns the strong redstone power emitted from the given position in the given direction.
+		COMMENT
```
Comment (EnnuiL) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+		COMMENT <p>
+		COMMENT NOTE: directions in redstone power related methods are backwards, so this method
+		COMMENT checks for power in the <i>opposite</i> direction of the one given.
+		ARG 1 pos
+		ARG 2 dir
+	METHOD m_vjljrkvs getFilteredEmittedRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;Z)I
+		COMMENT Returns the filtered redstone power emitted from the given position in the given direction.
+		COMMENT In 'strict' mode the filter only allows strong redstone power from redstone diodes (repeaters and comparators),
+		COMMENT while in 'lenient' mode it allows strong redstone power from any block as well as weak redstone power from redstone blocks.
+		COMMENT
```
Comment (EnnuiL) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]
hello! midnight ennui here; this looks good on a glance, but i do have nitpicks;
i'll review it properly tomorrow

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 7]
```diff
@@ -15,6 +15,10 @@ CLASS net/minecraft/unmapped/C_triydqro net/minecraft/block/AbstractBlock
 	METHOD <init> (Lnet/minecraft/unmapped/C_triydqro$C_xnkxsdfy;)V
 		ARG 1 settings
 	METHOD m_bhohbmwk getWeakRedstonePower (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)I
+		COMMENT Returns the weak redstone power this block emits in the given direction.
```
Comment (LambdAurora) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		COMMENT {@return the weak redstone power this block emits in the given direction}
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 8]
```diff
@@ -202,6 +206,10 @@ CLASS net/minecraft/unmapped/C_triydqro net/minecraft/block/AbstractBlock
 		ARG 4 projectile
 	METHOD m_xfdklwdd getVerticalModelOffsetMultiplier ()F
 	METHOD m_ydisyayk getStrongRedstonePower (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)I
+		COMMENT Returns the strong redstone power this block emits in the given direction.
```
Comment (LambdAurora) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		COMMENT {@return the strong redstone power this block emits in the given direction}
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 9]
```diff
@@ -0,0 +1,44 @@
+CLASS net/minecraft/unmapped/C_gpzjbzcw net/minecraft/world/RedstonePowerView
+	FIELD f_caqhuckp DIRECTIONS [Lnet/minecraft/unmapped/C_xpuuihxf;
+	METHOD m_dnkvwchb getEmittedRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)I
+		COMMENT Returns the redstone power emitted from the given position in the given direction.
```
Comment (LambdAurora) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
		COMMENT {@return the redstone power emitted from the given position in the given direction}
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+		COMMENT This is the highest value between power emitted by the block itself, and strong redstone power
+		COMMENT received from neighboring blocks if the block is a redstone conductor.
+		COMMENT <p>
+		COMMENT NOTE: directions in redstone power related methods are backwards, so this method
+		COMMENT checks for power in the <i>opposite</i> direction of the one given.
+		ARG 1 pos
+		ARG 2 dir
+	METHOD m_dsckesgu isEmittingRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)Z
+		COMMENT Returns whether redstone power is emitted from the given position in the given direction.
```
Comment (LambdAurora) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
		COMMENT {@return whether redstone power is emitted from the given position in the given direction}
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+		COMMENT <p>
+		COMMENT NOTE: directions in redstone power related methods are backwards, so this method
+		COMMENT checks for power in the <i>opposite</i> direction of the one given.
+		ARG 1 pos
+		ARG 2 dir
+	METHOD m_fwaivkfo isReceivingRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;)Z
+		COMMENT Returns whether the given position receives any redstone power from neighboring blocks.
+		ARG 1 pos
+	METHOD m_hizjdvsd getReceivedRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;)I
+		COMMENT Returns the redstone power the given position receives from neighboring blocks.
```
Comment (LambdAurora) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
		COMMENT {@return the redstone power the given position receives from neighboring blocks}
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+		ARG 1 pos
+	METHOD m_rgxdlzfx getReceivedStrongRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;)I
+		COMMENT Returns the strong redstone power the given position receives from neighboring blocks.
```
Comment (LambdAurora) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
		COMMENT {@return the strong redstone power the given position receives from neighboring blocks}
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+		ARG 1 pos
+	METHOD m_uegtusaf getEmittedStrongRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;)I
+		COMMENT Returns the strong redstone power emitted from the given position in the given direction.
```
Comment (LambdAurora) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
		COMMENT {@return the strong redstone power emitted from the given position in the given direction}
```

File: mappings/net/minecraft/world/RedstonePowerView.mapping [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+		COMMENT <p>
+		COMMENT NOTE: directions in redstone power related methods are backwards, so this method
+		COMMENT checks for power in the <i>opposite</i> direction of the one given.
+		ARG 1 pos
+		ARG 2 dir
+	METHOD m_vjljrkvs getFilteredEmittedRedstonePower (Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_xpuuihxf;Z)I
+		COMMENT Returns the filtered redstone power emitted from the given position in the given direction.
```
Comment (LambdAurora) [inline mappings/net/minecraft/world/RedstonePowerView.mapping]:
```suggestion
		COMMENT {@return the filtered redstone power emitted from the given position in the given direction}
```

**Review by LambdAurora — CHANGES_REQUESTED** [review]

**Comment by LambdAurora** [comment]
It's `{@return ...}` not `{@returns ...}` (this is why the build is failing)

**Comment by SpaceWalkerRS** [comment]
Woops! I'll push a quick fix.

**Review by LambdAurora — APPROVED** [review]

**Review by EnnuiL — APPROVED** [review]

**Pull request merged**